### PR TITLE
Simplify my proxy fix 9544

### LINKED ISSUE #9544: simplify myproxy credential name
**Impact of the new feature**
only affects CRAB.

**Is your feature request related to a problem? Please describe.**
currently submission from client  needs to go through same REST host as indicated in TaskWorker config. With multiple REST hosts and TW's in docker/k8s era this is not working. Also the numerical hash of the DN makes it always difficult to tell what's happening w/o adding a real extrasecurity value.

**Describe the solution you'd like**
credentail in myproxy should have a simple username, e.g. we can pass
to the WMCore Credential service a string like "CRAB" and credential name will be
CRAB_belforte etc. 

**Describe alternatives you've considered**
no other way is possible

**Additional context**

### PR DESCRIPTION
**Pull request by belforte**
Fixes #9544 

#### Status
ready

#### Description
1. if username is given in proxycfg, use it also for `logonRenewMyProxy()`
2. removes some fake error logging from `vomsExtensionRenewal`
Those functions are only used  by CRAB. You can safely merge and kindly add in next tag so that we can build a new TaskWorker with this and test in preprod.
I tested already in my VM.

#### Is it backward compatible (if not, which system it affects?)
YES 

#### Related PRs
#9548 was a partial fix for same issue

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by belforte** [comment]
Note: all failures reported by Jenkins are due to code parts which were not touched in this PR

File: src/python/WMCore/Credential/Proxy.py
Comment (amaltaro) [inline src/python/WMCore/Credential/Proxy.py]: there are a few typos in these comments, can you please fix them.

File: src/python/WMCore/Credential/Proxy.py
Comment (amaltaro) [inline src/python/WMCore/Credential/Proxy.py]: given that I see the same logic in different places in this module, how about we create a method to return the `credname` value? Just call that method every where you need to resolve the user credentials in the same format as: `sha1(self.userDN + "_" + self.myproxyAccount).hexdigest()`

**Review by amaltaro — CHANGES_REQUESTED** [review]
Stefano, please see the comments I have made in the code.

File: src/python/WMCore/Credential/Proxy.py
Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py]: OK. Also moved some lines around so comments are close to code they refer to.

**Comment by amaltaro** [comment]
test this please

File: src/python/WMCore/Credential/Proxy.py
Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py]: @amaltaro I started with the idea of making minimal modifications.

I do not like to introduce a `getCredname` method for something which is not exactly
a credential or anyhow adding a method I can't properly document.
Fact is that clearly different people added code to `Proxy.py` in uncoordinated way,
at times w/o fully understanding what they were doing. Hopefully one day we
will deprecate X509 and this stuff will all go. Hoping that day to be not too far,
I tried to avoid a major overhaul.

But  now that I learnt the inside of this and you made me think more about it, what about 
the following which is a truly minimal change to WMCore Proxy ?

we use `self.userName`  from the argument dictionary passed to
`Proxy.py` as "username" whenever a `myproxy-*` command is used which requires
the `-l username` option (only CRAB use that).
And leave it to the calling CRAB code to decide if to pass a human string or a hash of the DN+ the mysterious `self.myproxyAccount` which when you read this code you have no
clue what it should be ?

AFAIK only CRAB uses `-l username` option in myproxy-* (since it is only used when an indentity switch is needed) and nobody else was using the self.userName, at least not in WMCore repository.  Will also avoid using the name `credname` since this is not really the name of a credential, actually myproxy-* has a credential name argument, via the option `-k`.
I am not sure why the original developers used `-l username` instead of `-k credname` 

This requires to modify the changes I have done to CRAB already and a new round of test from my side, so do not hold your breath... but if you like it more, I'll do.

Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py]: sorry for very logg comment, let me summarize:
if I do what you suggested it will still be a non-understandable hack, which nobody but a CRAB expert can maintain/use, while saving only ~10 lines of code.
Since Proxy.py  will not be there in Run4 for sure, and hopefully gone sooner, I don't see the gain, but I can do it easily.

The alternative is more work for me, but a way to push some (but still not all) CRAB internals knowledge from Proxy.py back into CRAB land.

All in all, having slept on it, I'd rather do what you suggested as it gives me a place where to put a bunch of comments to help others to understand what's going on.

Comment (amaltaro) [reply src/python/WMCore/Credential/Proxy.py]: Stefano, your idea looks much clean indeed. However, it will also depend on how many clients/places you will have to apply such change. If not too much and you're willing to do so, please go ahead.

BTW, I don't think any of the WMCore services use this module. Even though we could, in order to renew my proxy that gets shipped with the jobs. Anyways, that's not important at the moment and I do not foresee investing time on that.

Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py]: looks like comments crossed. I'go for the simplest.
You will never need to use myproxy to refresh proxy for jobs, condor does it. As long as you have a single user, you only need to keep one valid proxy in each schedd, which I think you do already with the other methods in this class.

Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py]: I decided to follow initial suggestion, but use a method name which makes it clear it is a hack for CRAB and did my best to document it.
Hope you like it @amaltaro 

Do not merge yet. I need to test it now.

**Comment by amaltaro** [comment]
@belforte I assume you haven't completely tested it yet, right?

**Comment by belforte** [comment]
right, sorry. kid at home and all that..

**Comment by belforte** [comment]
This now ready to go from my side.

**Comment by belforte** [comment]
shall I remove the Do not merge yet label ?

File: src/python/WMCore/Credential/Proxy.py
Comment (amaltaro) [inline src/python/WMCore/Credential/Proxy.py:344]: I don't remember which pylint complaint is triggered here. A better logging format would be:
```
"Calculating hash of %s for credential name", self.userDN + "_" + self.myproxyAccount)
```

No need to update this patch for this line only though.

File: src/python/WMCore/Credential/Proxy.py
Comment (amaltaro) [inline src/python/WMCore/Credential/Proxy.py:411]: please fix the extra blank space between `self` and and the method name

File: src/python/WMCore/Credential/Proxy.py
Comment (amaltaro) [inline src/python/WMCore/Credential/Proxy.py:436]: same extra blank space here

**Review by amaltaro — CHANGES_REQUESTED** [review]
Stefano, sorry for the delay on this review.
Changes look good in general, but there are a few comments in the code that deserve to be fixed.

Once you make those changes, can you also please squash all these commits into a single one?

**Comment by belforte** [comment]
oh my.. how did I get those blank spaces in ?
@amaltaro, how do I squash other then while merging ? sorry..

File: src/python/WMCore/Credential/Proxy.py
Comment (belforte) [reply src/python/WMCore/Credential/Proxy.py:344]: done

**Comment by belforte** [comment]
I am getting lost in git rebase and company, maybe also becasue github is having problems atm.
May I close this and make a new PR with one single commit ?

**Pull request closed without merging**
